### PR TITLE
Create TypoScriptFrontendController in backend context

### LINKED ISSUE #23: TypeError in backend module "redirects" with "jumpurl" installed
TYPO3 10.4.1
redirects 10.4.1
jumpurl 8.0.0
PHP 7.4

Redirects with an external URL in field _target_ throw the following error in backend module _redirects_:

```
(1/1) TypeError

Return value of FoT3\Jumpurl\JumpUrlProcessor::getTypoScriptFrontendController() must be an instance of TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController, null returned

in [...]/public/typo3conf/ext/jumpurl/Classes/JumpUrlProcessor.php line 208

protected function getTypoScriptFrontendController(): TypoScriptFrontendController
{
    return $this->frontendController ?? $GLOBALS['TSFE'];
}
```

### PR DESCRIPTION
**Pull request by tobenschmidt**
Under certain circumstances it is neccessary to have a TypoScriptFrontendController available to avoid errors. For example the backend module of system extension redirects creates frontend links in backend context. This executes the urlProcessors hook of jumpurl but fails because there is no TypoScriptFrontendController. Method getTypoScriptFrontendController in class TYPO3\CMS\Frontend\Typolink\AbstractTypolinkBuilder provides some kind of workaround. This change is basically just copy and paste of this workaround.

Fixes #23

### CODE REVIEW

**Comment by christophlehmann** [comment]
Can someone review it / comment it?

File: Classes/JumpUrlProcessor.php [hunk 1]
```diff
@@ -205,7 +214,44 @@ protected function getTypoLinkParameter(array $configuration)
 
     protected function getTypoScriptFrontendController(): TypoScriptFrontendController
     {
-        return $this->frontendController ?? $GLOBALS['TSFE'];
+        if ($this->frontendController instanceof TypoScriptFrontendController) {
```
Comment (bmack) [inline Classes/JumpUrlProcessor.php:217]: This change does not check for `$GLOBALS['TSFE']` anymore if it ever existed.

**Comment by smichaelsen** [comment]
We successfully use this as a composer patch in our project.

File: Classes/JumpUrlProcessor.php (hunk 1 above)
Comment (gtitze) [reply Classes/JumpUrlProcessor.php:217]: I am not sure, but shouldn't it be sufficient to check for `$GLOBALS['TSFE']` in the constructor?
https://github.com/FriendsOfTYPO3/jumpurl/blob/6bb713275262c0ec18401aaae37653b70c9658ed/Classes/JumpUrlProcessor.php#L47-L51

**Comment by tbsschmdt** [comment]
I guess this has been solved with #27.

**Comment by bmack** [comment]
I think so too. closing this one. thanks for the feedback y'all.

**Pull request closed without merging**